### ISSUE TITLE
macOS compilation (fix)

### ISSUE THREAD

**Issue by ntapiam** [Tinob/Ishiiruka#76]
I was trying to build commit  [a5b0fcd](https://github.com/Tinob/Ishiiruka/commit/a5b0fcd5fbf15797a1dfb2f2ac02111e06ce680c) on macOS Sierra 10.12.4 with Apple clang-802.0.42, which failed with the following message:

```
[ 94%] Building CXX object Source/Core/DolphinWX/CMakeFiles/Dolphin.dir/Cheats/CheatsWindow.cpp.o
In file included from /Users/ntapia/Ishiiruka/Source/Core/DolphinWX/Cheats/CheatsWindow.cpp:37:
/Users/ntapia/Ishiiruka/Source/Core/DolphinWX/Frame.h:139:30: error: implicit instantiation of undefined template 'std::__1::array<TASInputDlg *, 8>'
        std::array<TASInputDlg*, 8> m_tas_input_dialogs{};
                                    ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__tuple:114:65: note: template is declared here
template <class _Tp, size_t _Size> struct _LIBCPP_TYPE_VIS_ONLY array;
                                                                ^
1 error generated.
make[2]: *** [Source/Core/DolphinWX/CMakeFiles/Dolphin.dir/Cheats/CheatsWindow.cpp.o] Error 1
make[1]: *** [Source/Core/DolphinWX/CMakeFiles/Dolphin.dir/all] Error 2
make: *** [all] Error 2
```

The fix was simply to add `#include <array>` at the top of `Source/Core/DolphinWX/Frame.h`.

**Comment by Tinob** [comment]
this is fixed in the latest version, Im interested in how ishiiruka works for mac users so feedback is welcome.

**Comment by ntapiam** [comment]
After successful compilation the first thing I can say is that sound does not work with OpenAL (no sound) or CoreAudio (jagged sound, low FPS).

I compiled using PulseAudio provided by Homebrew and all seems to work fine with this backend but I haven't done any extensive testing.

**Comment by ntapiam** [comment]
I did some more testing and I had to do another patching in the new version. Here's the patch:

```
diff --git a/CMake/DolphinPostprocessBundle.cmake b/CMake/DolphinPostprocessBundle.cmake
index 4d2eee175..d9d9f2710 100644
--- a/CMake/DolphinPostprocessBundle.cmake
+++ b/CMake/DolphinPostprocessBundle.cmake
@@ -37,7 +37,7 @@ set(extra_dirs "/usr/local/lib" "/lib" "/usr/lib")
 # BundleUtilities is overly verbose, so disable most of its messages
 function(message)
        if(NOT ARGV MATCHES "^STATUS;")
-               _message(${ARGV})
+               _message("${ARGV}")
        endif()
 endfunction()
```

**Comment by ntapiam** [comment]
So I went ahead and made a Homebrew formula that builds Stable and HEAD. Everything builds correctly now, automatically.

I've created a [Pull Request](https://github.com/Homebrew/homebrew-core/pull/12974) to get it integrated into Homebrew, I hope you don't mind.

**Comment by ntapiam** [comment]
So, the PR got rejected because you don't use tags. Also, it should produce a binary and not a .app bundle and I don't know if it's possible. In any case, I might start doing "regular" macOS builds and put them on a fork, maybe provide a Homebrew Cask after some testing.

Also, I noticed that for macOS (at least with Apple LLVM) the linker should also receive the flag `-lc++`. For the moment I pass it to CMake via -DCMAKE_EXE_LINKER_FLAGS manually.
There's also a problem with Gettext, which isn't found, at least for me. I had to install it via Homebrew and then pass the path manually.
